### ISSUE TITLE
ErrorException: Undefined index: siteId in  ... GlobalSet.php:38

### ISSUE THREAD

**Issue by daltonrooney** [jamesedmonston/graphql-authentication#68]
I just attempted an upgrade to Craft 3.7.11 and GraphQL-Authentication 1.11.3 and get the following error whenever I try to query a Global set that used to work: 

```
2021-09-07 23:04:32 [-][-][-][error][yii\base\ErrorException:8] yii\base\ErrorException: Undefined index: siteId in /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/jamesedmonston/graphql-authentication/src/resolvers/GlobalSet.php:38
Stack trace:
#0 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/web/ErrorHandler.php(75): yii\base\ErrorHandler->handleError()
#1 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/jamesedmonston/graphql-authentication/src/resolvers/GlobalSet.php(38): craft\web\ErrorHandler->handleError()
#2 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/gql/base/ElementResolver.php(86): jamesedmonston\graphqlauthentication\resolvers\GlobalSet::prepareQuery()
#3 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/gql/base/ElementResolver.php(39): craft\gql\base\ElementResolver::prepareElementQuery()
#4 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php(624): craft\gql\base\ElementResolver::resolveOne()
#5 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php(550): GraphQL\Executor\ReferenceExecutor->resolveFieldValueOrError()
#6 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php(1196): GraphQL\Executor\ReferenceExecutor->resolveField()
#7 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php(264): GraphQL\Executor\ReferenceExecutor->executeFields()
#8 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php(215): GraphQL\Executor\ReferenceExecutor->executeOperation()
#9 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/Executor/Executor.php(156): GraphQL\Executor\ReferenceExecutor->doExecute()
#10 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/GraphQL.php(162): GraphQL\Executor\Executor::promiseToExecute()
#11 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/webonyx/graphql-php/src/GraphQL.php(94): GraphQL\GraphQL::promiseToExecute()
#12 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/services/Gql.php(545): GraphQL\GraphQL::executeQuery()
#13 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/jamesedmonston/graphql-authentication/src/services/TokenService.php(306): craft\services\Gql->executeQuery()
#14 [internal function]: jamesedmonston\graphqlauthentication\services\TokenService->setActiveSchema()
#15 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/yiisoft/yii2/base/Event.php(312): call_user_func()
#16 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/yiisoft/yii2/base/Component.php(637): yii\base\Event::trigger()
#17 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/services/Gql.php(517): yii\base\Component->trigger()
#18 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/controllers/GraphqlController.php(172): craft\services\Gql->executeQuery()
#19 [internal function]: craft\controllers\GraphqlController->actionApi()
#20 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#21 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams()
#22 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/web/Controller.php(190): yii\base\Controller->runAction()
#23 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/yiisoft/yii2/base/Module.php(534): craft\web\Controller->runAction()
#24 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/web/Application.php(276): yii\base\Module->runAction()
#25 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/yiisoft/yii2/web/Application.php(104): craft\web\Application->runAction()
#26 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/craftcms/cms/src/web/Application.php(261): yii\web\Application->handleRequest()
#27 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/craft/vendor/yiisoft/yii2/base/Application.php(392): craft\web\Application->handleRequest()
#28 /home/hcps/webapps/hcps-api-dev/releases/5f59ccf968d65df0512c16a1653e71661ae70069/public/index.php(21): yii\base\Application->run()
#29 {main}
```

This site has been running GraphQL-Authentication since the early days so is it possible I missed a setting upgrade somewhere along the way?

**Comment by jamesedmonston** [comment]
@daltonrooney does it work if you resave the settings?

**Comment by daltonrooney** [comment]
No, I did try that, no luck. Is there something in my project.yaml file I should look for to see if they were saved successfully?

**Comment by jamesedmonston** [comment]
@daltonrooney the error was immediately obvious to me upon a second look! Have just pushed https://github.com/jamesedmonston/graphql-authentication/commit/aa695ec87274c57a01a01b08efc8234c86ece657 which fixes this

**Comment by jamesedmonston** [comment]
I've released `1.11.4` with this fix.
